### PR TITLE
[CAMEL-10853] Add support for some more properties

### PR DESCRIPTION
**Pull request by ribeaud**
OK, now it looks better...

### CODE REVIEW

File: camel-core/src/main/java/org/apache/camel/model/dataformat/CsvDataFormat.java [hunk 1]
```diff
@@ -163,6 +171,16 @@ protected void configureDataFormat(DataFormat dataFormat, CamelContext camelCont
         if (quoteMode != null) {
             setProperty(camelContext, dataFormat, "quoteMode", quoteMode);
         }
+        Stream.of("trim", "ignoreHeaderCase", "trailingDelimiter")
```
Comment (davsclaus) [inline camel-core/src/main/java/org/apache/camel/model/dataformat/CsvDataFormat.java]: Can you do like the existing code so its similar

**Comment by ribeaud** [comment]
Actually, I was rather thinking about migrating all this copy-paste property stuff into the Stream stuff... :) I never liked copy-paste. Sorry, no more time to work on this. Do what you want with it. Cheers, have a nice week-end and thanks for your time, work. christian.

**Comment by davsclaus** [comment]
Would you be able to work on this in next week perhaps?

Also mind that Camel is a large code base and its best to have code similar so users/maintainers can easier maintain/understand the code. Also we have to be able to backport code changes etc, for example 2.17.x is Java 7.

For example we have checkstyle to ensure code formatting is similar and so on. So in that light the java8 stream stuff cannot be done in one place and not all the other etc. 

If you dont want to work on this anymore, we may have to either close this PR, or merge it and make the changes ourselves. 

But Apache loves contributions so maybe you can make a change of heart and as a Camel user you can see this as a positive thing that you are actually able to contribute and get your patches into the code base.

**Comment by ribeaud** [comment]
> Also we have to be able to backport code changes etc, for example 2.17.x is Java 7.

OK, this is a good reason to NOT use **Java** streams. May I convert the stream into a more compatible array or list of properties?

> Also mind that Camel is a large code base and its best to have code similar so users/maintainers can easier maintain/understand the code.

Sorry Claus, but I CANNOT agree on that. To me, reading around 50 lines of _similar_ code is long, annoying and my brain stops spotting errors after a while because it is getting bored. And, as soon as I see so much _similar_ code, then I have some bells ringing trying to tell me that this could be done better resp. could be refactored. I used to lead a large public library once. So I know the challenges you are facing. But the hugeness of a library could NOT be an excuse for NOT improving things and writing beautiful code.

As **Camel** user, I feel obligated to do something. And I did it. If now you want me to duplicate even more _similar_ code, then I think you should do it. Otherwise, I could definitely find some time to refactor `configureDataFormat` and to make it shorter, more readable, finally quicker and easier to understand.

Kindly let me know. My humble respects,

christian

**Comment by davsclaus** [comment]
Okay thanks for the feedback. We will make the necessary code changes ourselves.

**Pull request closed without merging**